mlir-sycl/doc/developer/ContributeToSYCLMLIR.md

```diff
@@ -0,0 +1,102 @@
+# Contributing to SYCL-MLIR
+
+## 1) Fork the Intel LLVM repository
+
+We recommend contributors to work in their own forks of the Intel LLVM project. To create a fork:
+	
+  1) Visit https://github.com/intel/llvm and click the Fork button to establish a fork (e.g. intel-llvm)
+	2) Go to your newly created fork, and get the URL of you fork (green Code button):
+	
+	  https://github.com/<user>/intel-llvm.git    
+	  where <user> is your GitHub username.	
+
+## 2) Setup the forked repository
+
+Create your clone of the sycl-mlir branch (<user> stands in for your GitHub username):
+
+  ```sh
+  # Clone the forked repository, setup upstream repo.
+  git clone --recursive https://github.com/<user>/intel-llvm.git -b sycl-mlir
+  cd intel-llvm
+  git remote add upstream https://github.com/intel/llvm.git
+  # Ensure pushing to the upstream repository is not allowed.
+  git remote set-url --push upstream no_push
+  
+  # Verify that your remote is setup correctly, the command output should be:
+  # origin  https://github.com/<user>/intel-llvm.git (fetch)
+  # origin  https://github.com/<user>/intel-llvm.git (push)
+  # upstream  https://github.com/intel/llvm.git (fetch)
+  # upstream  no_push (push)
+  git remote -v
+  ```
+
+After the commands above, *upstream* refers to the original Intel LLVM repository while *origin* refers to your own fork of the repository.
+
+IMPORTANT: You should be careful to not commit changes your fork's *sycl-mlir* branch to avoid getting out of sync with the *sycl-mlir* branch in the *upstream* repository. 
+
+## 3) Keeping your fork up to date 
+
+To update your fork, first fetch the latest version of both your fork and the upstream repositories: 
+
+```sh 
+git fetch --all
+```
+
+Afterward update your fork  sycl-mlir branch:
+
+```sh
+git checkout sycl-mlir
+git merge origin/sycl-mlir
+git merge upstream/sycl-mlir
+git push origin sycl-mlir
+```
+
+Finally ensure that your fork *sycl-mlir* branch and the upstream one are identical (this should be the case if you never committed code into your fork sycl-mlir branch directly):
+
+```sh
+git diff upstream/sycl-mlir
+```
+
+## 4) Build and Testing
+
+Create a branch in your local fork:
+
+```sh
+git checkout sycl-mlir
+git checkout -b <name-of-your-branch> 
+```
+
+Configure and build the project:
+
+```sh
+python3 ./buildbot/configure.py --build-type Debug   
+python3 ./buildbot/compile.py   
+```
+
+You can now make local changes in your newly create branch. Once you have completed the changes please ensure all relevant lit tests pass:
+
+```sh
+ninja check-mlir-sycl 
+ninja check-cgeist
+ninja check polygeist-opt
+```
+
+## 5) Commit changes
+
+At this point you are ready to commit and push your changes. Ensure you sign your commits (git commit -s) before pushing them, also please prepend commit messages with `[SYCL-MLIR]` for easy identification:
```

Comment:
nit: We don't require signature in sycl branch (upstream) today. I would say it's optional.